### PR TITLE
Add Fressian Serde via clojure.data.fressian

### PR DESCRIPTION
**Pull request by bobby**
# Checklist

- [x] tests
- [x] updated [CHANGELOG](../CHANGELOG.md) (the "unreleased" section)

### CODE REVIEW

File: test/jackdaw/serdes/fressian_test.clj [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+             fressian/clojure-write-handlers)
+      fressian/associative-lookup
+      fressian/inheritance-lookup))
+
+(def read-handlers
+  (-> (merge {uri-tag (reify ReadHandler
+                        (read [_ reader tag component-count]
+                          (URI. (.readObject reader))))}
+             fressian/clojure-read-handlers)
+      fressian/associative-lookup))
```
Comment (gphilipp) [inline test/jackdaw/serdes/fressian_test.clj]: Not sure if that’s expected, but shouldn’t the read handler also use `fressian/inheritance-lookup` like the write handler ?

Comment (bobby) [reply test/jackdaw/serdes/fressian_test.clj]: @gphilipp I adapted that test from https://github.com/clojure/data.fressian/wiki/Creating-custom-handlers, and Stu didn't include the inheritance-lookup on the read side.  Inheritance is necessary when you've got an in-memory thingy and need to serialize it somehow, and somewhere in its inheritance hierarchy some superclass or interface knows how to be serialized.  It's less important when you've got a serialized thing and want to read it into a specific class, e.g. a `java.net.URI` in this case.

**Comment by bobby** [comment]
Thanks for approving, I look forward to a release with this added Serde!

**Comment by creese** [comment]
@bobby Please sign off on the commits. If you want to rebase these down to one commit, that's fine.

Instructions on how to sign off:
https://github.com/FundingCircle/jackdaw/blob/master/CONTRIBUTING.md#signoff-on-commits

**Comment by bobby** [comment]
@creese I've signed off on the commits, and the checks have passed again.

**Pull request merged**